common/metadata.go

```diff
@@ -0,0 +1,156 @@
+package common
+
+type UpsertMetadataAction string
+
+const (
+	// UpsertMetadataActionCreate indicates that the object/field was created.
+	UpsertMetadataActionCreate UpsertMetadataAction = "create"
+	// UpsertMetadataActionUpdate indicates that the object/field was updated.
+	UpsertMetadataActionUpdate UpsertMetadataAction = "update"
+	// UpsertMetadataActionNone indicates that the object/field was not changed.
+	UpsertMetadataActionNone UpsertMetadataAction = "none"
+)
+
+// UpsertMetadataParams represents parameters for upserting metadata.
+type UpsertMetadataParams struct {
+	// Maps object names to field definitions.
+	FieldsDefinitions map[string][]*FieldDefinition `json:"customFields"`
```

Comment:
Thinking about this some more, I think this should be `Fields` in order to match https://docs.withampersand.com/reference/objects-&-fields/get-object-metadata-for-connection, but the type can still be FieldDefinition, or FieldMetadata might make even more sense given that the method is called UpsertMetadata

Reply:
Also:
- I think `[string][]FieldMetadata` would be an easier type to deal with, and the json annotation still says `customFields`

Reply:
So this would become:

```
Fields map[string][]FieldMetadata `json:"Fields"`
```

Reply:
oops good catch. Sounds good.

Reply:
I think you missed my suggestion to turn this field name into `Fields`